IIS Reverse Proxy: Add preserveHostHeader step, + add extra examples (#168)
grafana/tutorials: content/tutorials/iis.md

# Use IIS with URL Rewrite as a Reverse Proxy

> If you want Grafana to be a subpath/subfolder under a website in IIS then the Application Request Routing (ARR) and URL Rewrite modules for ISS can be used to support this.

Reviewer: ```suggestion
To set Grafana as a subpath or subfolder under a website in IIS, use the Application Request Routing (ARR) and URL Rewrite modules for ISS for support.
```

Author: Hi Brenda, I deliberately changed `subpath or subfolder` to use a slash because they are the same thing.  Using `or` is misleading/misdirecting for the reader. otherwise other changes have no impact.

Example:

- Parent site: http://yourdomain.com:8080
- Grafana: http://localhost:3000

Grafana as a subpath: http://yourdomain.com:8080/grafana

Other Examples:

> - The parent site could also look like http://localhost:8080 if the application was only served on the local server.

Reviewer: ```suggestion
- If the application is only served on the local server, the parent site can also look like http://localhost:8080. 
```

Reviewer: Served on the local server sounds odd. I am not sure of the exact context, but perhaps we could find a better word?

Author: That's the best way to described it so there is no ambiguity.  The objective should not be perfect english (as the last one was not amazing in this regard) it should be to ensure the user can't misinterpret the instructions.

> - If your domain is served using https on port 443, and thus the port is not normally entered in the address of your site, then the need to specify a port for the parent site in the configuration steps below can be eliminated.

Reviewer: ```suggestion
- If your domain is served using https on port 443 and the port is not normally entered in the address of your site, you do not need to specify a port for the parent site in the configuration steps below.
```

 

## Setup

> If you have not already done it, install the URL Rewrite module for IIS.

Reviewer: ```suggestion
Install the URL Rewrite module for IIS.
```

>  - Download and install the URL Rewrite module for IIS: https://www.iis.net/downloads/microsoft/url-rewrite

Reviewer: ```suggestion
 - Download and install the URL Rewrite module for IIS: https://www.iis.net/downloads/microsoft/url-rewrite.
```

> You will also need the Application Request Routing (ARR) module for IIS for proxy forwarding

Reviewer: ```suggestion
Ensure you have the Application Request Routing (ARR) module for IIS for proxy forwarding.
```

>  - Download and install ARR module for IIS: https://www.iis.net/downloads/microsoft/application-request-routing

Reviewer: ```suggestion
 - Download and install ARR module for IIS: https://www.iis.net/downloads/microsoft/application-request-routing.
```

## Grafana Config

The Grafana config can be set by creating a file named/editing the existing file named `custom.ini` in the `conf` subdirectory of your Grafana installation. See the [installation instructions](http://docs.grafana.org/installation/windows/#configure) for more details.

> Using the example from above, if the subpath is `grafana` (you can set this to whatever is required) and the parent site is `yourdomain.com:8080`, then you would add this to the `custom.ini` config file:

Reviewer: ```suggestion
Using the example above, if the subpath is `grafana` (you can set this to whatever is required) and the parent site is `yourdomain.com:8080`, add this to the `custom.ini` config file:
```

 ```bash
[server]
domain = yourdomain.com:8080
root_url = %(protocol)s://%(domain)s/grafana/
```

Restart the Grafana server after changing the config file.

Configured address to serve Grafana: http://yourdomain.com:8080/grafana

------

> If you already had a subpath on your domain you would configure it like this

Reviewer: ```suggestion
If you already have a subpath on your domain, configure it as follows:
```

{{< figure src="/static/img/docs/tutorials/IIS_add_inbound_rule.png"  max-width="800px" >}}

5. Create an Inbound Rule for the website with the following settings:
  - pattern: `grafana(/)?(.*)`  (if you have customised the subpath that will be used, use that instead of `grafana`)
  - check the `Ignore case` checkbox
  - rewrite URL set to `http://localhost:3000/{R:2}`
  - check the `Append query string` checkbox
  - check the `Stop processing of subsequent rules` checkbox

{{< figure src="/static/img/docs/tutorials/IIS_url_rewrite.png"  max-width="800px" >}}

> 6. If your version of Grafana is greater than 8.3.5 you will also need to configure the reverse proxy to preserve host headers

Reviewer: ```suggestion
6. If your version of Grafana is greater than 8.3.5, you also need to configure the reverse proxy to preserve host headers.
```
